ext/spl/spl_dllist.stub.php

```diff
@@ -5,29 +5,22 @@
 class SplDoublyLinkedList implements Iterator, Countable, ArrayAccess, Serializable
 {
     /**
-     * @param mixed $index
-     * @param mixed $value
+     * @param resource|string|int|float|bool $index
```

Comment:
I think these should be changed to proper `int` parameters instead.

Reply:
Yeah, it's definitely a good idea! There's a slight problem though: `SplDoublyLinkedList::offsetSet` will "push" the value when the index is `null`  (like `$array[] = "";`), so I guess I have to leave this function as-is :(

Just to be sure, I shouldn't do the conversion for `SplDoublyLinkedList::offsetExists()`, right (it also uses `spl_offset_convert_to_long`)? I'm tempted to do so, but the weak semantics also has its advantages, that's why I feel that I should leave it alone as well.